hazelcast/src/main/java/com/hazelcast/map/IMap.java

```diff
@@ -70,28 +70,6 @@
  * {@code entrySet}, return an <b>immutable</b> collection clone of the values.
  * The collection is <b>NOT</b> backed by the map, so changes to the map are
  * <b>NOT</b> reflected in the collection.</li>
- * <li>Since Hazelcast is compiled with Java 1.6, we can't override default
- * methods introduced in later Java versions, nor can we add documentation
- * to them. Methods, including but not limited to {@code computeIfPresent},
- * may behave incorrectly if the value passed to the update function is
- * modified in-place and returned as a result of the invocation.
- * You should create a new value instance and return it as a result.
- * <p>
- * For example, following code fragment will behave incorrectly and will
- * enter an infinite loop:
- * <pre>
- * map.computeIfPresent("key", (key, value) -&gt; {
- *     value.setSomeAttribute("newAttributeValue");
- *     return value;
- * });
- * </pre>
- * It should be replaced with:
- * <pre>
- * map.computeIfPresent("key", (key, value) -&gt; {
- *     return new ObjectWithSomeAttribute("newAttributeValue");
- * });
- * </pre>
- * </li>
```

Comment:
Removing this part of the documentation as Hz is already compiled with Java 8, and we have a plan to support Java 8 methods (#14913). 
Happy to revert this change in case I should keep this until we actually address #14913

Reply:
Yes, let's remove that once we actually have the implementations ready and covered by tests.